### PR TITLE
Added flow for renaming ceph pools

### PR DESCRIPTION
**Pull request by shtripat**
tendrl-bug-id: Tendrl/api#168
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

File: tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    RequestStateError
+from tendrl.commons import objects
+from tendrl.ceph_integration.objects.pool import Pool
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons.objects import AtomExecutionFailedError
+
+
+class Rename(objects.BaseAtom):
+    obj = Pool
```
Comment (r0h4n) [inline tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: this is not required, why do you keep adding it your atoms :pager:

File: tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    def run(self):
+        pool_id = self.parameters['Pool.pool_id']
+        attrs = {}
+        attrs['name'] = self.parameters.get('Pool.poolname')
+        Event(
+            Message(
+                priority="info",
+                publisher=NS.publisher_id,
+                payload={
+                    "message": "Renaming the pool with pool-id %s."
```
Comment (r0h4n) [inline tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: mention the new name and old name along with id

File: tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        resp = crud.update("pool", pool_id, attrs)
+        try:
+            crud.sync_request_status(resp['request'])
+        except RequestStateError as ex:
+            Event(
+                Message(
+                    priority="info",
+                    publisher=NS.publisher_id,
+                    payload={
+                        "message": "Failed to rename pool %s."
```
Comment (r0h4n) [inline tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: mention old name and failed new name

File: tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                )
+            )
+            return False
+
+        Event(
+            Message(
+                priority="info",
+                publisher=NS.publisher_id,
+                payload={
+                    "message": "Pool %s successfully renamed" %
```
Comment (r0h4n) [inline tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: mention old name and new name

**Review by r0h4n — CHANGES_REQUESTED** [review]

File: tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py (hunk 1 above)
Comment (shtripat) [reply tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: Will remove that. I feel a clean is required for other objects as well.

File: tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py (hunk 2 above)
Comment (shtripat) [reply tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: For old name we would need to fetch the pool from etcd. Anyway will add new name at least.

File: tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py (hunk 3 above)
Comment (shtripat) [reply tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: For old name we would need to fetch the pool from etcd. Anyway will add new name at least.

File: tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py (hunk 4 above)
Comment (shtripat) [reply tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: Same as ^^

File: tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        )
+
+        crud = Crud()
+        resp = crud.update("pool", pool_id, attrs)
+        try:
+            crud.sync_request_status(resp['request'])
+        except RequestStateError as ex:
+            Event(
+                Message(
+                    priority="info",
```
Comment (anmolsachan) [inline tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: Priority can be error here.

Comment (shtripat) [reply tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: done

File: tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py (hunk 1 above)
Comment (r0h4n) [reply tendrl/ceph_integration/objects/pool/atoms/rename/__init__.py]: Sure, send a PR

File: tendrl/ceph_integration/objects/pool/flows/rename_pool/__init__.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+from tendrl.commons import flows
+from tendrl.ceph_integration.objects.pool import Pool
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+
+
+class RenamePool(flows.BaseFlow):
+    obj = Pool
```
Comment (r0h4n) [inline tendrl/ceph_integration/objects/pool/flows/rename_pool/__init__.py]: obj = Pool is not required man

**Review by r0h4n — CHANGES_REQUESTED** [review]

File: tendrl/ceph_integration/objects/pool/flows/rename_pool/__init__.py (hunk 6 above)
Comment (shtripat) [reply tendrl/ceph_integration/objects/pool/flows/rename_pool/__init__.py]: My bad. Will remove that.

**Review by r0h4n — APPROVED** [review]

**Pull request merged**